qutebrowser/utils/urlmatch.py

```diff
@@ -177,6 +178,13 @@ def _init_host(self, parsed):
             assert self._host is None
             return
 
+        if not utils.raises(ValueError, ipaddress.IPv6Address, parsed.netloc[1:-1]):
```

Comment:
I think you could remove this, and instead raise a `ParseError` with `url.errorString()` if `url.isValid()` is false after setting the host.

However, you should probably only do this `if parsed.netloc.startswith('['):` which kinda mirrors the Chromium code.